addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -4,103 +4,163 @@ odoo.define('website_sale_delivery.checkout', function (require) {
-                disabledReasons.carrier_selection = true;
-                $payButton.data('disabled_reasons', disabledReasons);
+        if (carriers.length > 0) {
+            const carrierChecked = carriers.filter(e =>e.checked)
+            if (carrierChecked.length === 0) {
+                const payButton = document.querySelector('button[name="o_payment_submit_button"]');
+                payButton? payButton.disabled = true : null;
+            } else {
+                carrierChecked[0].click();
             }
-            $carriers.filter(':checked').click();
-        }
-
-        // Asynchronously retrieve every carrier price
-        _.each($carriers, function (carrierInput, k) {
-            self._showLoading($(carrierInput));
-            self._rpc({
-                route: '/shop/carrier_rate_shipment',
-                params: {
-                    'carrier_id': carrierInput.value,
-                },
-            }).then(self._handleCarrierUpdateResultBadge.bind(self));
-        });
+        }
 
-        return this._super.apply(this, arguments);
+        await this._getCurrentLocation();
+        await _.each(carriers, async carrierInput => {
+            this._showLoading((carrierInput));
+            await this._handleCarrierUpdateResult(carrierInput)
+        });
+        if (this._super && typeof(this._super.apply)==="function") {
+          return this._super.apply(this, arguments);
+        }
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
+    /**
+     * @private
+     */
+    _getCurrentLocation: async function () {
+        const data = await this._rpc({
+            route: "/shop/access_point/get",
+        })
+        const carriers = document.querySelectorAll('.o_delivery_carrier_select')
+        for (let carrier of carriers) {
+            const deliveryType = carrier.querySelector('input[type="radio"]').getAttribute("delivery_type");
+            const deliveryName = carrier.querySelector('label').innerText;
+            const showLoc = carrier.querySelector(".o_show_pickup_locations");
+            if (!showLoc) {
+                continue;
+            }
+            const orderLoc = carrier.querySelector(".o_order_location");
+            if (data[deliveryType + '_access_point'] && data.delivery_name == deliveryName) {
+                orderLoc.querySelector(".o_order_location_name").innerText = data.name
+                orderLoc.querySelector(".o_order_location_address").innerText = data[deliveryType + '_access_point']
+                orderLoc.parentElement.classList.remove("d-none");
+                showLoc.classList.add("d-none");
+                break;
+            } else {
+                orderLoc.parentElement.classList.add("d-none");
+                showLoc.classList.remove("d-none");
+            }
+        }
+    },
```

Comment:
Hello, I randomly noticed this working on something unrelated.

Since this is called in the `start()` of the public widget, this is firing this RPC on every pages of the ecommerce, while it's actually not doing anything with it unless you are on the payment page (where `.o_delivery_carrier_select` matches something and so you enter the loop).

Could you fix that to shortcircuit the method and not fire the RPC if you are not on the expected page containing the DOM element you need?